src/Microsoft.AspNetCore.JsonPatch/Internal/JsonPatchErrorHelper.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.JsonPatch.Exceptions;
+
+namespace Microsoft.AspNetCore.JsonPatch.Internal
+{
+    internal static class JsonPatchErrorHelper
```

Comment:
Using the word helper anywhere in a PR is a surefire way to get me to object to it. 

I'd suggest something like this.

```
public static class ErrorReporter
{
    public static readonly Action<JsonPatchError> Default = (error) => throw new JsonPatchException(error);
}
```

Then everywhere else is just 
```
errorReporter = errorReporter ?? ErrorReporter.Default
```